crates/generate_artifacts/src/generate_artifacts.rs

```diff
@@ -721,37 +842,60 @@ fn write_param_type_from_selection<TOutputFormat: OutputFormat>(
-                            schema,
-                            &linked_field.selection_set,
-                            object,
-                            nested_client_field_imports,
-                            loadable_fields,
-                            indentation_level,
-                            link_fields,
-                            updatable_fields,
-                        );
-                    query_type_declaration.push_str(&format!("readonly {}: ", name_or_alias,));
-                    query_type_declaration.push_readonly(&print_javascript_type_declaration(
-                        &associated_data
+                    let type_annotation =
+                        associated_data
                             .type_name
                             .clone()
-                            .map(&mut |_| &parameter_type),
-                    ));
-                    query_type_declaration.push_updatable(&print_javascript_type_declaration(
-                        &associated_data
-                            .type_name
-                            .clone()
-                            .map(&mut |_| &updatable_parameter_type),
-                    ));
-                    query_type_declaration.push_str(",\n");
+                            .map(&mut |output_type_id| {
+                                let object_id = output_type_id;
+                                let object = schema.server_field_data.object(object_id);
+                                generate_client_field_updatable_data_type(
+                                    schema,
+                                    &linked_field.selection_set,
+                                    object,
+                                    nested_client_field_imports,
+                                    loadable_fields,
+                                    indentation_level,
+                                    link_fields,
+                                    updatable_fields,
+                                )
+                            });
+
+                    match linked_field.associated_data.selection_variant {
+                        ValidatedIsographSelectionVariant::Loadable(_) => {
+                            panic!("@loadble server fields  are not supported yet")
+                        }
+                        ValidatedIsographSelectionVariant::Updatable => {
+                            *updatable_fields = true;
+
+                            query_type_declaration.push_str(&format!(
+                                "get {}(): {},\n",
+                                name_or_alias,
+                                print_javascript_type_declaration(&type_annotation),
+                            ));
+                            let setter_type_annotation = associated_data
+                                .type_name
+                                .clone()
+                                .map(&mut |_| "{ link: Link }");
```

Comment:
this is gonna be pretty funny when we actually model unions and we have `Foo | Bar` -> `{ link: Link } | { link: Link }` lol.

Anyway, all good, we'll deal with that at some future point